### DISCUSSION TITLE
Wave attenuation in Elastic Media

### DISCUSSION THREAD

**Discussion by pharshaad** [idaholab/moose#18720] [Q&A Modules: Solid mechanics] [answered]
Hi all,

I'm studying 1D wave propagation in an elastic material. I'm sending a sinusoidal pulse with 5MHz frequency by displacing one of the boundaries using FuncionDirichletBC (similar to wave_newmark.i example in tests/dynamics folder). What I'm observing is that the wavefront gets attenuated quite a bit even though the Newmark time integration parameters are \beta = 0.25 and \gamma = 0.5. This effect seems to be less pronounced in lower frequencies though. I was wondering what mechanism is behind this that I'm missing and is there a way to avoid it?

Thanks,
Ari

**Comment by hugary1995** [comment]
Are you suggesting the Newmark beta time integrators is giving you incorrect solution? If so, could provide us with the correct/analytical solution to benchmark with?

If you are asking why wave attenuation is more pronounced with higher frequency inputs, then I honestly don't know (this isn't a moose question TBH). Tagging @cbolisetti in case he has some suggestions in this regard.

**Reply by cbolisetti** [reply]
This sounds like a meshing issue. In wave propagation, you need to make sure that your mesh is fine enough to capture all the frequencies you want. Higher frequencies have a smaller wavelength, and if your mesh is too coarse it may not be able to resolve these wavelengths. In wave propagation for seismic waves in soil, we typically follow a rule of thumb of having 10 elements per wavelength for the highest frequency you want to capture. We refine the mesh based on this number. That should ensure that the frequencies you are aiming for will propagate without attenuation. Let us know if that works. 

--
Sorry for the autocorrect errors earlier. Was typing on the phone while I was in a rush.

**Comment by ngrilli** [comment, marked as answer]
If you attach the input file, it will be easier to answer.

As suggested by others, try to refine the mesh and see if the effect goes away,
make sure there are no damping terms as described here:
https://mooseframework.inl.gov/modules/tensor_mechanics/Dynamics.html

Make sure your time step is appropriate,
start with: 
dt = element size / propagation velocity / 10

I have done wave propagation in moose and never
observed damping unless I introduced it on purpose.

Nicolò

**Reply by pharshaad** [reply]
Thanks @hugary1995 , @cbolisetti, and @ngrilli for the suggestions. It was the issue of time step size not being chosen small enough. 

Cheers,
Ari
